lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -108,6 +110,14 @@ defmodule ExUnit.Assertions do
     vars = collect_vars_from_pattern(left)
     pins = collect_pins_from_pattern(left, Macro.Env.vars(__CALLER__))
 
+    left_pattern = escape_quoted(:pattern, left)
+
+    pattern_vars =
+      vars
+      |> Enum.map(fn {var, _, _} -> var end)
+      |> Enum.uniq()
+      |> Enum.map(&{&1, :ex_unit_unbound_var})
```

Comment:
You cannot trust variable names. Take this code:

```elixir
var = quote(do: right)
quote do: {right, unquote(var)}
```

They have the same name but they are not the same.

Reply:
Is the meta data ever significant for variables, or is it just the first element and the last element that are significant?

Reply:
A variable is uniquely identified by this: `fn {var, meta, context} -> {var, meta[:counter] || context} end`.

Reply:
Thanks.  I updated the vars.  I had a go at the pins, but was not able to get a pinned var from another context - is that true, or just a lack of imagination on my part?

Reply:
@jeremyowensboggs it seems to be intended https://github.com/elixir-lang/elixir/commit/6931b9d7be8b7f88439bd2434fb1256b58d1f2a9